File: conformance/tests/if-else/if-else.tsop.go
package contract

import "tsop"

type IfElse struct {
	tsop.SmartContract
	Limit tsop.Int `tsop:"readonly"`
}

func (c *IfElse) Check(value tsop.Int, mode tsop.Bool) {
	result := tsop.Int(0)
	if mode {
		result = value + c.Limit
	} else {
		result = value - c.Limit
	}
	tsop.Assert(result > 0)
}

File: conformance/tests/if-else/if-else.tsop.rs
use tsop::prelude::*;

#[tsop::contract]
struct IfElse {
    #[readonly]
    limit: Int,
}

#[tsop::methods(IfElse)]
impl IfElse {
    #[public]
    fn check(&self, value: Int, mode: bool) {
        let mut result: Int = 0;
        if mode {
            result = value + self.limit;
        } else {
            result = value - self.limit;
        }
        assert!(result > 0);
    }
}
